src/activities/menu/Menu.qml

```diff
@@ -218,12 +235,31 @@ ActivityBase {
                     }
 
                     function selectCurrentItem() {
-                        particles.burst(10)
-                        ActivityInfoTree.filterByTag(modelData.tag)
-                        ActivityInfoTree.filterLockedActivities()
-                        ActivityInfoTree.filterEnabledActivities()
-                        menuActivity.currentTag = modelData.tag
-                        section.currentIndex = index
+
+
+                               if(modelData.tag === "search")
+                               {
+                                   section.currentIndex = index
+                                   menuActivity.currentTag = modelData.tag
+                                   searchbar.visible = true;
```

Comment:
if possible, set the variables directly in qml instead of using js. For example, in this case, in searchBar item you should have:
visible: modelData.tag === "search". This way, it will automatically update